### ISSUE TITLE
Difference in behaviour between IDE and Terminal using the same rust install, .cargo and toolchain

### ISSUE THREAD

**Issue by manoadamro** [intellij-rust/intellij-rust#9013]
## Environment

* **IntelliJ Rust plugin version:** 0.4.173.4714-222
* **Rust toolchain version:**  1.62.0
* **IDE name and version:** 2022.1.3  &  2022.2 EAP
* **Operating system:** Ubuntu (wsl)

## Problem description

Sync is trying (and failing) to update from crates.io. What's particularly odd is, the IDE is configured to use the same rust install as my terminal, and crates.io index is up to date. I can run `build`, `test`, etc from the terminal just fine but for some reason intellij wants to update it again, hangs for an hour or so, then exists with the following error:

```
Execution failed (exit code 101).
C:\Windows\system32\wsl.exe --distribution Ubuntu --exec /bin/sh -c "export PATH=\"/home/adam/.cargo/bin:$PATH\" && export RUST_BACKTRACE=short && export TERM=ansi && cd /home/adam/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/test && /home/adam/.cargo/bin/cargo '+stable-x86_64-unknown-linux-gnu' metadata --verbose --format-version 1 --all-features"
stdout :     Updating crates.io index
warning: spurious network error (2 tries remaining): failed to connect to github.com: Network is unreachable; class=Os (2)
warning: spurious network error (1 tries remaining): failed to connect to github.com: Network is unreachable; class=Os (2)
error: failed to get `cfg-if` as a dependency of package `test v0.0.0 (/home/adam/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/test)`

Caused by:
  failed to load source for dependency `cfg-if`

Caused by:
  Unable to update registry `crates-io`

Caused by:
  failed to fetch `https://github.com/rust-lang/crates.io-index`

Caused by:
  failed to connect to github.com: Network is unreachable; class=Os (2)

stderr : 

Execution failed (exit code -1).
C:\Windows\system32\wsl.exe --distribution Ubuntu --exec /bin/sh -c "export PATH=\"/home/adam/.cargo/bin:$PATH\" && export RUST_BACKTRACE=short && export TERM=ansi && cd /home/adam/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core && /home/adam/.cargo/bin/cargo '+stable-x86_64-unknown-linux-gnu' metadata --verbose --format-version 1 --all-features"
stdout :     Updating crates.io index
warning: spurious network error (2 tries remaining): failed to connect to github.com: Network is unreachable; class=Os (2)
```

From the terminal, I managed to run `cargo build`, which has updated the crates.io index at the exact same path that the IDE is trying to update. using the same rustc and toolchain.

---

I could just ignore all the IDE frills and stick with the terminal, but this sync is popping up all the time and blocking the lock file, so I'm constantly having to stop the sync, run `cargo clean` and get my terminal command in before sync opens again.

In the meantime, I'll try a few more older versions of the IDE to see if any work,

The `failed to get `cfg-if` as a dependency of package `test v0.0.0` is interesting, I'm not actually using the `test` crate anymore, could this be a dependency of a dependency?

---

Possibly related: I have the lalrpop plugin also installed, but it has an error next to it saying `Requires plugin 'org.rust.lang' to be installed`. I have the "Rust" (this plugin) installed and have restated both the IDE and the machine several times. I can't seem to find anything else relating to `core,lang.rust`, is there something else I need to install for this to work?

## Steps to reproduce

- Install rust
- Install intellij
- Configure intellij to use rust from wls Ubuntu.
- Reopen Intellij and let it try to sync,

**Comment by manoadamro** [comment]
Downgrading to `2021.2.4` has allowed me to use the configured crates.io index instead of always starting from scratch. I haven't tried `2021.3.3` but it happened in version `2022.1.3` and later.
